keyboards/nullbitsco/nibble/matrix.c

```diff
@@ -0,0 +1,167 @@
+/* Copyright 2018 Jack Humbert
```

Comment:
This entire file can probably be significantly simplified by swapping to `CUSTOM_MATRIX = lite` in your `rules.mk`.
All things such as debounce and the like are abstracted away, and the implementation just needs to provide the raw gathering of the matrix data.

You'd only need to provide:
```c
void matrix_init_custom(void) { /* ... */ }
bool matrix_scan_custom(matrix_row_t current_matrix[]) {
    bool has_changed = false;
    /* ... */
    return has_changed;
}
```